src/bmerge.c

```diff
@@ -280,21 +280,21 @@ void bmerge_r(int xlowIn, int xuppIn, int ilowIn, int iuppIn, int col, int thisg
         /* Regular roll=TRUE|+ve|-ve */                                                           \
         if ((( roll>0.0 && (!lowmax || xlow>xlowIn) && (xupp<xuppIn || !uppmax || rollends[1]))   \
           || ( roll<0.0 && xupp==xuppIn && uppmax && rollends[1]) )                               \
-          && ( isinf(rollabs) || ((LOWDIST)-(TYPE)rollabs <= 0) ))                                \
+          && ( isinf(rollabs) || ((LOWDIST)-(TYPE)rollabs <= (TYPE)1e-6) ))                       \
```

Comment:
Could you explain this change?

Reply:
When TYPE is int or int64_t, (TYPE)1e-6 casts down to 0 as before. When TYPE is double, it is 1e-6 as before. So it's just a change to make the macro-ization work.  The bigger change vs master on this line is the `isinf()` first.  That's for efficiency to avoid the comparison happening at all, but also to simplify the solution for #1007 and reduce an extra comparison. All tests pass (there is no change to tests.Rraw in this PR at all) so I'm assuming no behavior has changed. If it has then we are missing a test for that behavior.